src/components/productList/ProductListFlatList.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {
+  ActivityIndicator,
+  FlatList,
+  RefreshControl,
+  Text,
+  View,
+} from 'react-native';
+import {styles} from './styles';
+import {COLORS} from '../../utils/colors';
+import ProductListCard from '../productListCard';
+import STRING_UTILS from '../../utils/StringUtils';
+import {getIncludedImageById} from '../../api/ImageApi';
+import {useNavigation} from '@react-navigation/native';
+
+type ProductListFlatListProp = {
+  data: {};
+  included: {};
+  onRefresh: any;
+  isRefreshing: boolean;
+};
+const ProductListFlatList = ({
+  data,
+  included,
+  onRefresh,
+  isRefreshing,
+}: ProductListFlatListProp) => {
+  const navigation = useNavigation();
+  if (!data) {
```

Comment:
It seems, that we'd better use ternary here, to impro readibility.
Ideally, `loading` functionality is better to be moved to it's own context and interacted throug it. Thus providing loading state for multiple sources at once. Loading indicator component, in such case, can be moved to a independent instance.